packages/cli/src/services/check-parser/parser.ts

```diff
@@ -16,45 +16,33 @@ type Module = {
   npmDependencies: Array<string>
 }
 
-enum FileExtensions {
-  JS = '.js',
-  TS = '.ts',
-  PACKAGE = '/package.json',
```

Comment:
I refactored this because the `check-parser` tests were failing https://github.com/checkly/checkly-cli/actions/runs/4322164114/jobs/7544240734#step:8:10
```
FAIL unit src/services/check-parser/__tests__/check-parser.spec.ts (5.074 s)
  ● dependency-parser - parser() › should handle JS file with dependencies

    expect(received).toEqual(expected) // deep equality

    - Expected  - 2
    + Received  + 2

      Array [
        "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\dep1.js",
        "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\dep2.js",
        "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\dep3.js",
    +   "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\module-package/package.json",
        "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\module-package\\main.js",
    -   "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\module-package\\package.json",
    -   "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\module\\index.js",
    +   "D:\\a\\checkly-cli\\checkly-cli\\packages\\cli\\src\\services\\check-parser\\__tests__\\check-parser-fixtures\\simple-example\\module/index.js",
      ]

      19 |     const parser = new Parser(defaultNpmModules)
      20 |     const { dependencies } = parser.parse(toAbsolutePath('entrypoint.js'))
    > 21 |     expect(dependencies.map(d => d.filePath).sort()).toEqual([
         |                                                      ^
      22 |       toAbsolutePath('dep1.js'),
      23 |       toAbsolutePath('dep2.js'),
      24 |       toAbsolutePath('dep3.js'),

      at Object.<anonymous> (src/services/check-parser/__tests__/check-parser.spec.ts:21:54)
```

The dependencies on `package.json` and `index.js` were being properly detected, but the path that was returned was using a mix of `\` and `/` separators. 

I could simply update the tests to loosen the assertions. I don't think that we have an actual bug here - Windows seems to have some support for mixing the `/` and `\` separators. On the other hand, I think that it could make sense to stick with the `\` separator on Windows.

Changing check-parser to use a platform dependent path separator required some extra changes, though. We can only use constant strings in TS-enums, so I needed to re-work the code to stop using enums.